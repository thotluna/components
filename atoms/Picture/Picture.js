import React from 'react'
import PropTypes from 'prop-types'
import { styles } from '.'
import withStyles from '../../hocs/withStyles'

export const Picture = ({
  src,
  alt,
  sources,
  width,
  height,
  isRounded,
  withBorder,
  getStyles,
}) => {
  return (
    <picture className={getStyles('picture')}>
      {sources &&
        sources.map((source) => (
          // eslint-disable-next-line react/jsx-key
          <source
            key={source.src}
            srcSet={source.src}
            media={source.media}
          ></source>
        ))}
      <img
        className={getStyles('image', {
          'is-rounded': isRounded,
          'with-border': withBorder,
        })}
        style={{ maxWidth: width, height: isRounded ? width : height }}
        src={src}
        alt={alt}
        role={!alt || alt === '' ? 'presentation' : ''}
      />
    </picture>
  )
}

Picture.propTypes = {
  src: PropTypes.string.isRequired,
  getStyles: PropTypes.func.isRequired,
  alt: PropTypes.string,
  sources: PropTypes.arrayOf(
    PropTypes.shape({
      src: PropTypes.string,
      media: PropTypes.string,
    })
  ),
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  isRounded: PropTypes.bool,
  withBorder: PropTypes.bool,
}

Picture.defaultProps = {
  alt: '',
  width: 200,
  height: 'auto',
  isRounded: false,
  withBorder: false,
}

export default withStyles(styles)(Picture)
