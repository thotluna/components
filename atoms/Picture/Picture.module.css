.picture {
  overflow: hidden;
}

.image {
  display: block;
  width: 100%;
  object-fit: cover;
}

.is-rounded {
  border-radius: var(--border-radius-full);
}

.with-border {
  border: var(--picture-border);
}
