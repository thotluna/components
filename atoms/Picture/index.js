export { default } from './Picture'
export { options } from './constants'
export { default as styles } from './Picture.module.css'
