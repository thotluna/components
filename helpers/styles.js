import classNames from 'classnames'

export const getDynamicClasses = (cssModule = {}, props = {}, classes) => {
  return classes.reduce((classesObject, classKey) => {
    const propValue = props[classKey]
    const className = cssModule[`${classKey}-${propValue}`]

    return className && propValue
      ? { ...classesObject, [className]: propValue }
      : classesObject
  }, {})
}

export const getModuleClasses = (cssModule, classKey) => {
  return (cssModule && cssModule[classKey]) || classKey
}

export const getObjectClasses = (cssModule, object) => {
  return Object.keys(object).reduce((classes, classKey) => {
    const className = cssModule[classKey]
    return className ? { ...classes, [className]: object[classKey] } : classes
  }, {})
}

export const getClasses = (cssModule = {}) => (props) => (...args) => {
  const allClass = args.map((arg) => {
    if (Array.isArray(arg)) {
      const dinamicClasses = getDynamicClasses(cssModule, props, arg)
      // console.log('>>>dinamic', dinamicClasses)
      return dinamicClasses
    } else if (typeof arg === 'string') {
      const moduleClass = getModuleClasses(cssModule, arg)
      // console.log('>>>module', moduleClass)
      return moduleClass
    } else if (typeof arg === 'object') {
      const objectClass = getObjectClasses(cssModule, arg)
      // console.log('>>>object', objectClass)
      return objectClass
    }
  })
  // console.log(allClass)

  const result = classNames(allClass)

  // console.log(result)
  return result
}
