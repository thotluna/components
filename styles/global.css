@import url('https://fonts.googleapis.com/css2?family=Architects+Daughter&family=Montserrat:ital,wght@0,400;0,700;1,100&display=swap');
html,
body {
  height: 100%;
  font-family: var(--font-family-title);
  font-size: var(--font-size-base);
}

#root {
  height: 100%;
}