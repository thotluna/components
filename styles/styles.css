@import './minireset.min.css';
@import './tokens.css';
@import './global.css';
@import './variables.css';
