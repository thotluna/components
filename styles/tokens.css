:root {
  --color-brand-blue-angel: #b4d1fe;
  --color-brand-light-slate-blue: #7b61ff;
  --color-brand-mercury: #e5e5e5;
  --color-brand-mid-gray: #595f68;
  --color-brand-mist-blue: #676e79;
  --color-brand-charcoal-gray: #3b3f45;
  --color-brand-blossom-ping: #feb4ef;
  --color-brand-shocking-ping: #fb0ecc;
  --color-base-black: #000;
  --color-base-white: #fff;
  --color-base-transparent: transparent;
  --spacing-none: 0;
  --spacing-xs: 5;
  --spacing-sm: 15;
  --spacing-md: 20;
  --spacing-lg: 40;
  --media-query-min-width-sm: 576px;
  --media-query-min-width-md: 768px;
  --media-query-min-width-lg: 992px;
  --media-query-min-width-xl: 1200px;
  --media-query-max-width-xs: 575.98px;
  --media-query-max-width-sm: 767.98px;
  --media-query-max-width-md: 991.98px;
  --media-query-max-width-lg: 1199px;
  --font-family-sans: 'Open Sans', arial, helvetica, sans-serif;
  --font-family-title: 'Architects Daughter', arial, helvetica, sans-serif;
  --font-size-base: 10px;
  --font-size-xs: 1.1rem;
  --font-size-sm: 1.3rem;
  --font-size-md: 1.8rem;
  --font-size-lg: 2rem;
  --font-size-xl: 2.1rem;
  --font-size-2xl: 6rem;
  --font-weight-hairline: 100;
  --font-weight-normal: 400;
  --font-weight-bold: 700;
  --line-height-none: 1;
  --line-height-tight: 1.25;
  --line-height-snug: 1.375;
  --line-height-normal: 1.5;
  --line-height-relaxed: 1.625;
  --line-height-loose: 2;
  --letter-spacing-tighter: -0.05em;
  --letter-spacing-tight: -0.025em;
  --letter-spacing-normal: 0;
  --letter-spacing-wide: 0.025em;
  --letter-spacing-wider: 0.05em;
  --letter-spacing-widest: 0.1em;
  --box-shadow-none-t: none;
  --box-shadow-xs-t: 0 3px 6px rgba(0, 0, 0, 0.05);
  --box-shadow-sm-t: 0 4px 5px rgba(0, 0, 0, 0.07);
  --border-radius-none: 0;
  --border-radius-sm: 10px;
  --border-radius-md: 20px;
  --border-radius-lg: 30px;
  --border-radius-full: 9999px;
  --border-width-none: 0;
  --border-width-thin: 1px;
  --border-width-thick: 2px;
  --opacity-0: 0;
  --opacity-25: 0.25;
  --opacity-50: 0.5;
  --opacity-75: 0.75;
  --opacity-100: 1;
  --z-index-0: 0;
  --z-index-10: 10;
  --z-index-20: 20;
  --z-index-30: 30;
  --z-index-40: 40;
  --z-index-50: 50;
  --z-index-auto: auto;
  --color-background: #3b3f45;
  --color-primary: #b4d1fe;
  --color-secondary: #feb4ef;
  --color-font-base: #b4d1fe;
  --color-font-primary: #3b3f45;
  --color-font-error: #fb0ecc;
  --button-base-background: transparent;
  --button-base-color: #b4d1fe;
  --button-base-color-border: #b4d1fe;
  --button-primary-background: #b4d1fe;
  --button-primary-color: #3b3f45;
  --button-primary-color-border: #b4d1fe;
  --button-secondary-background: #feb4ef;
  --button-secondary-color: #3b3f45;
  --button-secondary-color-border: #b4d1fe;
  --container-max-width: 800px;
}
