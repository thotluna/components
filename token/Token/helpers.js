import toKebabCase from '../../utils/toKebalCase'

export const getTokenItems = (tokens, tokenKey) => (group) => {
  if (group) {
    return Object.keys(tokens[group]).map((key) => ({
      label: `--${toKebabCase(tokenKey)}-${group}-${toKebabCase(key)}`,
      value: tokens[group][key],
    }))
  }

  return Object.keys(tokens).map((key) => ({
    label: `--${toKebabCase(tokenKey)}-${toKebabCase(key)}`,
    value: tokens[key],
  }))
}
