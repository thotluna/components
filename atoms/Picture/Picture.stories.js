import { Picture } from './Picture'
import styles from './Picture.module.css'
import { getTemplate } from '../../helpers/storybook'

const Template = getTemplate(Picture, styles)

const args = {
  src: 'https://picsum.photos/200/200',
  alt: 'It one image',
  width: 150,
  height: 'auto',
  isRounded: false,
  withBorder: false,
}

export default {
  title: 'Atoms/Picture',
  component: Picture,
  args,
  argTypes: {
    src: { control: 'text' },
    alt: { control: 'text' },
    source: { control: 'object' },
    width: { control: 'number' },
    height: { control: 'number' },
  },
  parameters: { __sb: { mh: 300 } },
}

export const Default = Template.bind({})
Default.args = { ...args }

export const ImageDecorative = Template.bind({})
ImageDecorative.args = {
  ...args,
  alt: '',
}

export const ImageRounded = Template.bind({})
ImageRounded.args = {
  ...args,
  isRounded: true,
  width: 200,
  height: 200,
}

export const ImageWithBorder = Template.bind({})
ImageWithBorder.args = {
  ...args,
  withBorder: true,
}

export const ImageRounderWithBorder = Template.bind({})
ImageRounderWithBorder.args = {
  ...args,
  withBorder: true,
  isRounded: true,
  width: 200,
  height: 200,
}
