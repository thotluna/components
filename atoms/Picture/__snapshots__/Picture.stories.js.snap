// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Picture Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": 300,
    }
  }
>
  <picture
    className="picture"
    style={
      Object {
        "height": "auto",
        "maxWidth": 150,
      }
    }
  >
    <img
      alt="It one image"
      className="image"
      role=""
      src="https://picsum.photos/200/200"
    />
  </picture>
</div>
`;

exports[`Storyshots Atoms/Picture Image Decorative 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": 300,
    }
  }
>
  <picture
    className="picture"
    style={
      Object {
        "height": "auto",
        "maxWidth": 150,
      }
    }
  >
    <img
      alt=""
      className="image"
      role="presentation"
      src="https://picsum.photos/200/200"
    />
  </picture>
</div>
`;

exports[`Storyshots Atoms/Picture Image Rounded 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": 300,
    }
  }
>
  <picture
    className="picture"
    style={
      Object {
        "height": 200,
        "maxWidth": 200,
      }
    }
  >
    <img
      alt="It one image"
      className="image is-rounded"
      role=""
      src="https://picsum.photos/200/200"
    />
  </picture>
</div>
`;

exports[`Storyshots Atoms/Picture Image Rounder With Border 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": 300,
    }
  }
>
  <picture
    className="picture"
    style={
      Object {
        "height": 200,
        "maxWidth": 200,
      }
    }
  >
    <img
      alt="It one image"
      className="image is-rounded with-border"
      role=""
      src="https://picsum.photos/200/200"
    />
  </picture>
</div>
`;

exports[`Storyshots Atoms/Picture Image With Border 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": 300,
    }
  }
>
  <picture
    className="picture"
    style={
      Object {
        "height": "auto",
        "maxWidth": 150,
      }
    }
  >
    <img
      alt="It one image"
      className="image with-border"
      role=""
      src="https://picsum.photos/200/200"
    />
  </picture>
</div>
`;
